ui/pages/permissions-connect/permissions-connect.component.js

```diff
@@ -113,8 +118,10 @@ export default class PermissionConnect extends Component {
 
     if (history.location.pathname === connectPath && !isRequestingAccounts) {
       ///: BEGIN:ONLY_INCLUDE_IN(flask)
-      if (isSnap) {
+      if (isSnap && !isUpdatingSnap) {
```

Comment:
Couldn't this be detected by the fact that the snap is already installed?

Reply:
Otherwise, maybe we need an enum instead of a flag for future use

Reply:
hmm yeah or I can just check for `permissionsRequest.newPermissions`